source/AMDSnDIndex.h

```diff
@@ -256,14 +256,14 @@ bool AMDSnDIndex::operator!=(const AMDSnDIndex& other) const {
 
 quint32 AMDSnDIndex::product() const {
 	quint32 rv = 1;
-	for(quint8 mu=rank_-1; mu>=0; --mu)
+	for(qint8 mu=rank_-1; mu>=0; --mu)
```

Comment:
I realize that there was a warning for this, but what was the rational of changing the for loop to use `qint8` rather than changing `rank_` to be `quint8`?